### PR TITLE
fix: Fix issue mr relation, issue stream, get next issue state.

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Fix issue mr relation, issue stream, get next issue state.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjk3LDEyODYsMTI4MCwxMjUwXSwiYXNzaWduZWUiOlsiMTAwMTA3MyJdfQ%3D%3D&id=320240&iterationID=1280&pId=0&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Fix issue mr relation, issue stream, get next issue state.            |
| 🇨🇳 中文    |     修复事项日志 和获取下一个状态         |

#### Need cherry-pick to release versions?
release/2.2

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/apps/dop/providers/issue/stream/service.go [hunk 1]
```diff
@@ -195,7 +195,7 @@ func (s *CommentIssueStreamService) PagingIssueStreams(ctx context.Context, req
 		return nil, err
 	}
 	iss := make([]*pb.IssueStream, 0, len(issueStreams))
-	for _, v := range issueStreams {
+	for i, v := range issueStreams {
 		is := &pb.IssueStream{
```
Comment (sfwn) [inline internal/apps/dop/providers/issue/stream/service.go:199]: define `v := v` is better

File: internal/apps/dop/providers/issue/core/provider.go [hunk 2]
```diff
@@ -160,10 +160,19 @@ type IssuePermProjectID interface {
 	GetProjectID() uint64
 }
 
+type IssuePermProjectIDint64 interface {
```
Comment (sfwn) [inline internal/apps/dop/providers/issue/core/provider.go:163]: strange name ends with `int64`.
maybe `IssuePermProjectIDFetcher` is better.

File: internal/apps/dop/providers/issue/stream/service.go (hunk 1 above)
Comment (shuofan) [reply internal/apps/dop/providers/issue/stream/service.go:200]: done

File: internal/apps/dop/providers/issue/core/provider.go (hunk 2 above)
Comment (shuofan) [reply internal/apps/dop/providers/issue/core/provider.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**